expression/schema.go

```diff
@@ -153,13 +153,23 @@ func (s *Schema) IsUniqueKey(col *Column) bool {
 // ColumnIndex finds the index for a column.
 func (s *Schema) ColumnIndex(col *Column) int {
 	for i, c := range s.Columns {
-		if c.FromID == col.FromID && c.Position == col.Position {
+		if c.Position == col.Position {
 			return i
 		}
 	}
 	return -1
 }
 
+// FindColumnByName finds a column by its name.
+func (s *Schema) FindColumnByName(name string) *Column {
+	for _, col := range s.Columns {
+		if col.ColName.L == name {
```

Comment:
Should we take the DBName and TableName into consideration?

Reply:
No, this is only used for datasource which only have one table.